### PR TITLE
feat(Tree): Add tests

### PR DESCRIPTION
**Pull request by leko91**

### CODE REVIEW

File: src/components/Tree/Tree.cy.tsx [hunk 1]
```diff
@@ -34,37 +66,244 @@ const TreeComponent = ({ onSelect }: { onSelect?: (id: Nullable<string>) => void
...
+        const onSelectStub = cy.stub().as('onSelectStub');
+
+        cy.mount(<TreeComponent onSelect={onSelectStub} selectedIds={['1']} />);
+
+        cy.get(TREE_ITEM_ID).first().click();
+
+        cy.get('@onSelectStub').should('have.been.calledOnce');
+    });
+
+    it('expands/shrinks the correct items when uncontrolled', () => {
```
Comment (GoranRibic) [inline src/components/Tree/Tree.cy.tsx:128]: as uncontrolled component

File: src/components/Tree/Tree.cy.tsx [hunk 2]
```diff
@@ -34,37 +66,244 @@ const TreeComponent = ({ onSelect }: { onSelect?: (id: Nullable<string>) => void
...
 
-        cy.get(TREE_ITEM_ID).should('have.length', 3);
         cy.get(TREE_ITEM_TOGGLE_ID).first().click();
         cy.get(TREE_ITEM_ID).should('have.length', 8);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+        cy.get(TREE_ITEM_ID).should('have.length', 3);
+    });
+
+    it('calls onExpand when controlled by the Tree', () => {
```
Comment (GoranRibic) [inline src/components/Tree/Tree.cy.tsx:138]: same as previous

Comment (GoranRibic) [reply src/components/Tree/Tree.cy.tsx:138]: please fix all the test names when the same name pattern occurs

File: src/components/Tree/Tree.cy.tsx [hunk 3]
```diff
@@ -34,37 +66,244 @@ const TreeComponent = ({ onSelect }: { onSelect?: (id: Nullable<string>) => void
...
+    it('allows adding an item dynamically to the Tree', () => {
+        const newNode: TreeItemMock = {
+            id: '4',
+            label: 'New Added Item',
+        };
+        const rootItemsLength = treeItemsMock.length;
+
+        cy.mount(<TreeComponent />).then(({ rerender }) => {
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength);
+            treeItemsMock.push(newNode);
```
Comment (GoranRibic) [inline src/components/Tree/Tree.cy.tsx:274]: aren't you affecting other test with this mutation of treeItemsMock?

File: src/components/Tree/Tree.cy.tsx [hunk 4]
```diff
@@ -34,37 +66,244 @@ const TreeComponent = ({ onSelect }: { onSelect?: (id: Nullable<string>) => void
...
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength + 1);
+        });
+    });
+
+    it('allows removing an item dynamically to the Tree', () => {
+        const rootItemsLength = treeItemsMock.length;
+
+        cy.mount(<TreeComponent />).then(({ rerender }) => {
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength);
+            treeItemsMock.pop();
```
Comment (GoranRibic) [inline src/components/Tree/Tree.cy.tsx:285]: Same as above

File: src/components/Tree/Tree.cy.tsx (hunk 3 above)
Comment (leko91) [reply src/components/Tree/Tree.cy.tsx:274]: The pushed item gets popped in the next test so it should be fine, no?

**Comment by imoutaharik** [comment]
@anxobotana @fabriciomendonca Same here, Can I close this PR? Or are you retaking it?

**Comment by anxobotana** [comment]
closing in favor of https://github.com/Frontify/fondue/pull/1482

**Pull request closed without merging**